routes/summary.js

```diff
@@ -0,0 +1,43 @@
+const express = require('express');
+const { DECENNIAL_LATEST_TABLE_FULL_PATH } = require('../special-calculations/data/constants');
+
+const router = express.Router();
+
+/*
+ * Helper function to calculate totals
+ * @param{Object} data - the data to calculate totals with
+ * @param{Object} variableNames - the variable names to calculate totals for
+ */
+function summarizeData(data, variableNames) {
+  var summarizedData = {};
+  variableNames.forEach((variableName) => { summarizedData[variableName] = 0; });
+  data.forEach((item) => { summarizedData[item.variable] += item.value });
+  return summarizedData;
+}
+
+/*
+ * Checks that the summary query parameter is a valid summary type
+ * @param{string} summary - The summary type
+ * @returns{Boolean}
+ */
+function isInvalidSummary(summary) {
+  const validSummaryNames = ['decennial', 'acs'];
+  if (validSummaryNames.includes(summary)) return false;
+  return true;
+}
+
+router.get('/:summary/:summaryvars/:geoids/', async (req, res) => {
+  const { app } = req;
+
+  const { summary, summaryvars, geoids: _geoids } = req.params;
+
+  const varString = "'" + summaryvars.split(',').join("','") + "'";
+  const idString = "'" + _geoids.split(',').join("','") + "'";
+
+  if (isInvalidSummary(summary)) res.status(400).send({ error: 'Invalid summary name. Summary must be acs or decennial' });
+  const summaryData = await app.db.query(`SELECT * FROM ${DECENNIAL_LATEST_TABLE_FULL_PATH} WHERE "geoid" IN (${idString}) and "variable" IN (${varString})`);
```

Comment:
@godfreyyeung I'm curious what you think about this approach. I have a few thoughts but this code as-is is pretty much what I had in mind

* We might want to tweak the words used to refer to the route params here to be more inline with the existing "survey" routes. For example, `:summary` here is meant to be either `decennial` or `acs`, just like the `:survey` part of the `/survey/:survey/:geotype/:geoid` route you designed the last time we worked on PFF. Maybe we can come up with a good route path for this new endpoint that nests under `/survey` somehow?
* To me this route is pretty "generic"(which is great!) so the term "summary" feels like its leaking in from our use case for it on the front end. Potentially another reason to maybe nest it under "/survey". Things like `:summaryvars` could become just `:variables` or `:variableIds` or something along those lines
* Thoughts on how David is receiving and parsing out the route params that are essentially "arrays"? I've never really found a universal "best practice" way of doing that but I think how it's done here is totally fine for our purposes.
* Normally I would say we shouldn't be making calls to the db directly in the controller but it's not like this app really has a separate "Model" layer for us to build upon. I don't really mind doing it right in the controller like this for now, as long as we all understand why that's not really best practice.

@dhochbaum-dcp we might want to make some tweaks to the overall API design based on my comments above but in general this is pretty much exactly what I had in mind. One question I had about this bit of code - If the intention here is that the API expects either `decennial` or `acs` for `:summary`, why does the sql query seem to hard code the decennial table? I know our use case only requires us to look at decennial but if the route params support ACS, I think the query should as well. I would also be cool with just designing this API to only look at that table for now since it's what we need and it wouldn't be hard to upgrade it to also look at ACS in the future, I just think the implementation needs to be one or the other.

A couple things I would like you to tweak;
* Doing `SELECT *` in queries is generally something to avoid. We should explicitly select only the properties we want and need so that query behaviors don't change if the underlying table schema changes and we aren't pulling more data than we need to
* Can you wrap that `app.db.query` in a try catch? You can return a 500 response code in the catch if something goes wrong executing the query

Reply:
I will add ACS in for consistency, and will wrap the query in a try/catch.  

How much do you want to minimize the data?  The table only has 4 columns, year (which I assume is all 2020), geoid, variable, and value.  I had the endpoint return the entire set of results, along with the totals, to keep it as "generic" as possible.  If we don't want to pull more data than we need to, we don't even need to bother pulling the individual rows - I could replace the SQL with the query `SELECT "variable",SUM("value") FROM ${table} WHERE "geoid" IN (${idString}) and "variable" IN (${varString}) GROUP BY "variable"` and just get the totals we need for the map summary box.  I'm happy to make as generic or specific as you would like, just let me know what you consider to be the sweet spot.

Reply:
Oh good point. for this endpoint we can go with that `SUM("value")` query (I think the corresponding column in the ACS tables is "estimate"). That keeps the response smaller and simpler and if we ever needed to get the individual rows for lists of variables and geoids, that would be a separate endpoint.